dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -282,7 +282,7 @@ def next_script_level
 
   def user_or_session_level
     if current_user
-      current_user.next_unpassed_progression_level(@script).try(:or_next_progression_level)
+      current_user.next_unpassed_visible_progression_level(@script).try(:or_next_progression_level)
```

Comment:
is the `or_next_progression_level` bit the part that handles the "we're on the last level in the script, so there _is_ no next progression level" logic?

Reply:
Yes, but it was plagued by a similar problem. If you complete the last level without completing all of the previous ones, and click continue on the unit overview page, you'll end up on the first level of the script _even if it's hidden_, in which case you get the "your teacher wasn't expecting you here" message. In the latest change, I removed `or_next_progression_level` and account for the last level in a progression in `next_unpassed_visible_progression_level`.